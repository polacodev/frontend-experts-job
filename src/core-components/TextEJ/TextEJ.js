import React from 'react';
import { Text } from 'native-base';

const TextEJ = ({ type = '', children }) => {
  const typeOfText = (element) => {
    switch (element) {
      case 'normal':
        return <Text numberOfLines={1}>{children}</Text>;
      case 'note':
        return (
          <Text note numberOfLines={1}>
            {children}
          </Text>
        );
      default:
        return null;
    }
  };

  return typeOfText(type);
};

export default TextEJ;
