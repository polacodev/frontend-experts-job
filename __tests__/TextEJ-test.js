import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';

import TextEJ from '../src/core-components/TextEJ/TextEJ';
jest.useFakeTimers();
describe('<TextEJ />', () => {
  test('renders <TextEJ /> correctly', () => {
    const tree = renderer
      .create(<TextEJ type="normal" children="test label example" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should be no render the component due the fake props', async () => {
    const { container } = render(
      <TextEJ type="fake type" children="test no-label example" />,
    );
    expect(container.firstChild).toBeUndefined();
  });
});
