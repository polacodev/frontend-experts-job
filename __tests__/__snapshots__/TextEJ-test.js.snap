// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextEJ /> renders <TextEJ /> correctly 1`] = `
<Text
  numberOfLines={1}
  style={
    Object {
      "color": "#000",
      "fontFamily": "System",
      "fontSize": 16,
    }
  }
  uppercase={false}
>
  test label example
</Text>
`;
